python_modules/dagster/dagster/core/host_representation/external_data.py

```diff
@@ -697,6 +699,8 @@ def __new__(
         depended_by: Sequence[ExternalAssetDependedBy],
         compute_kind: Optional[str] = None,
         op_name: Optional[str] = None,
+        op_names: Optional[Sequence[str]] = None,
```

Comment:
What do you think about removing the `op_name` field above and only using `node_names`? (since a node in the graph could be a nested graph)

I think it would reduce confusion since there could be:
- no node names (source asset node)
- one node name (@asset or @multi_asset)
- multiple node names (graph)

and having one field be the source of truth on the type of node seems easier to handle

Reply:
agree that it's not in an ideal state right now, and op_name should be removed (this is basically pending removing all of its references in JS land, which might take a bit).

I do think we need two separate fields here though -- sometimes, a graph backed asset can have just a single node inside of it, meaning that having a single node name is not always evidence enough that we're working with a GBA. i suppose we could do some fanciness to detect that the single node name had a dot in it, but i think it ends up being more convenient to store these separately.